pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/Loaders.java

```diff
@@ -92,13 +92,12 @@
       // Add, remove or replace default columns
       // Add column min/max values
       // NOTE: this step may modify the segment metadata.
-      try (SegmentPreProcessor preProcessor = new SegmentPreProcessor(indexDir, indexLoadingConfig,
-          schema)) {
+      try (SegmentPreProcessor preProcessor = new SegmentPreProcessor(indexDir, indexLoadingConfig, schema)) {
         preProcessor.process();
       }
 
       // Load the metadata again since converter and pre-processor may have changed it
-      SegmentMetadataImpl segmentMetadata = new SegmentMetadataImpl(indexDir);
+      SegmentMetadataImpl segmentMetadata = new SegmentMetadataImpl(indexDir, segmentVersion);
```

Comment:
rename segmentVersion to segmentVersionToLoad and metadataVersion to diskVersion or something similar.

Previous PR also removed shouldConvertSegment() method call that wrapped logic  for code readability. MEthod for single condition may seem overkill but those encapsulate logic making calling code more robust and readable. And those methods don't add cost. Please avoid making such changes. Prefer readability